io/teca_shape_file_mask.h

```diff
@@ -12,7 +12,13 @@
 TECA_SHARED_OBJECT_FORWARD_DECL(teca_shape_file_mask)
 
 /// Generates a valid value mask defined by regions in the given ESRI shape file
-/** This algorithm is a source and has no inputs. A
+/** The teca_shape_file_mask generates a mask on a 2D mesh where mesh points
+ * are assigned a value of 1 where they fall inside any one of the polygons
+ * defined in the named ESRI shape file, and 0 everywhere else. The input mesh
```

Comment:
Based on the code here, it looks like it is assumed that the polygons in the shape file are in [the EPSG:4326 projection](https://spatialreference.org/ref/epsg/4326/) (or WSG84).  This should be reported by the CRS variable in the shapefile, so it should be possible to check.  Maybe this is checked somewhere that I'm not immediately seeing though?

If I'm right about this, I suggest changing the documentation to indicate this (there are many shapefiles out in the wild that are in other projections, which would not work here) and possibly also implementing a check on the CRS value.  In a future pair of PRs, it might be worth considering (1) adding [GDAL](https://gdal.org/) to the superbuild, and (2) [using GDAL to transform](https://gdal.org/api/ogrspatialref.html) between the shapefiles coordinate system and EPSG:4326 coordinate system.

Reply:
You're right, except that we can handle both -180 to 180 and 0 to 360 in lon.  I've encountered a number of shape files in other coordinate systems and with our current implementation there's no indication of a problem except a mask of all zeros. Then I've used other tools to dump the shapefile inorder to investigate further.

The library we're currently using http://shapelib.maptools.org/ may not support the CRS field. In fact I can find no mention of the term "coordinate system" in the shapefile format specification http://dl.maptools.org/dl/shapelib/shapefile.pdf Where is the CRS field documented?

Thanks for the pointer to gdal, added an issue for this #744